### PR TITLE
[mergeBot] port arg parsing logic to argparse+shlex

### PR DESCRIPTION
**Pull request by suo**
This brings a few benefits:
- shlex ensures that the command-line parsing is POSIX-compliant, which
conforms to people's expectations of what is allowed on a command line
(as an example, single-quoted strings should work).
- argparse is a more declarative way of specifying arguments, which
eliminates a lot of handwritten code for, e.g. required arguments,
choice allowlists, generating a help message, and so on.
- argparse also is more familiar to PyTorch in terms of syntax users, as
it is the conventional thing that Python programs use. (e.g. single dash
for short form, double dash for long form)

### CODE REVIEW

**Comment by suo** [comment]
This PR also deletes the old regex-based logic (well almost all of it, we still need to port the pull request review stuff, which doesn't support the CLI either). Since we said we were going to cut over on june 1, might as well do it here :P

File: torchci/test/mergeBot.test.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ describe("merge-bot", () => {
 
   test("merge this comment on issue triggers confused reaction", async () => {
```
Comment (malfet) [inline torchci/test/mergeBot.test.ts:67]: Please change test name

File: torchci/test/mergeBot.test.ts [hunk 2]
```diff
@@ -50,11 +50,11 @@ describe("merge-bot", () => {
 
   test("quoted merge/revert command no event", async () => {
     const merge_event = require("./fixtures/issue_comment.json");
-    merge_event.payload.comment.body = "> @pytorchbot merge this";
+    merge_event.payload.comment.body = "> @pytorchbot merge";
     const revert_event = require("./fixtures/issue_comment.json");
-    revert_event.payload.comment.body = "> @pytorchbot revert this";
```
Comment (malfet) [inline torchci/test/mergeBot.test.ts:55]: Can we add a negative test that "merge this"/"revert this" would result in some sort of error

**Review by malfet — APPROVED** [review]

File: torchci/lib/bot/cliParser.ts [hunk 3]
```diff
@@ -1,103 +1,89 @@
...
+  action: "store_true",
+  help: "Merge when all tests pass",
+});
+merge.add_argument("-f", "--force", {
+  action: "store_true",
+  help: "Merge without checking anything",
+});
+
+// Revert
+const revert = commands.add_parser("revert", {
```
[the remark this answers is not in the document]
Comment (suo) [reply torchci/lib/bot/cliParser.ts:39]: Please noooo lol

**Pull request merged**